pkg/cmd/run.go

```diff
@@ -967,9 +967,22 @@ func (o *runCmdOptions) getTargetPath() string {
 }
 
 func (o *runCmdOptions) uploadFileOrDirectory(platform *v1.IntegrationPlatform, item string, integrationName string, cmd *cobra.Command, integration *v1.Integration) error {
-	uri := parseFileURI(item)
-	o.RegistryOptions = uri.Query()
-	localPath, targetPath := uri.Path, o.getTargetPath()
+	var targetPath, localPath string
+	if strings.HasPrefix(item, "http://") || strings.HasPrefix(item, "https://") {
+		uri, err := url.Parse(item)
+		if err != nil {
+			return err
+		} else if localPath, err = downloadDependency(o.Context, *uri); err != nil {
+			return err
+		}
+		// Remove the temporary file
+		defer os.Remove(localPath)
```

Comment:
Where is `targetPath` initialised in this case?

Reply:
@tadayosi [line 975](https://github.com/apache/camel-k/pull/3644/files#diff-d9b5c5a6245814d60e99f92a71feb162ec636c09f5787321970e26ef80fb736bR975)

Reply:
OK, but then what happens if user specifies some txt file other than jar or pom via http(s)?  In the below code `targetPath` seems to be still used. Doesn't it break the cli?

Reply:
I proposed a specific syntax with pipe as separator to be able to set the options